internal/configs/resource.go

```diff
@@ -647,6 +576,210 @@ func decodeReplaceTriggeredBy(expr hcl.Expression) ([]hcl.Expression, hcl.Diagno
+		valDiags := eval.DecodeExpression(attr.Expr, ident, &r.Managed.CreateBeforeDestroy)
+		diags = append(diags, valDiags...)
+		r.Managed.CreateBeforeDestroySet = true
+	}
+
+	if attr, exists := lcContent.Attributes["prevent_destroy"]; exists {
+		ident := StaticIdentifier{Module: eval.call.addr, Subject: fmt.Sprintf("%s.%s.lifecycle.%s", r.Type, r.Name, attr.Name), DeclRange: attr.Range}
+		valDiags := eval.DecodeExpression(attr.Expr, ident, &r.Managed.PreventDestroy)
+		diags = append(diags, valDiags...)
+		r.Managed.PreventDestroySet = true
+	}
+
+	lcContent = r.Managed.lifecycleBody
+
+	attr, exists := lcContent.Attributes["ignore_changes"]
+
+	if !exists {
+		return diags
+	}
+
+	return diags.Extend(r.decodeLifecycleIgnoreChanges(eval, attr))
+}
+
+func (r *Resource) decodeLifecycleIgnoreChanges(eval *StaticEvaluator, attr *hcl.Attribute) hcl.Diagnostics {
+	var diags hcl.Diagnostics
+
+	// ignore_changes can either be a list of relative traversals
+	// or it can be just the keyword "all" to ignore changes to this
+	// resource entirely.
+	//   ignore_changes = [ami, instance_type] // hclsyntax.TulpleConsExpr *ExprList* hclsyntax.ScopeTraversalExpression
+	//   ignore_changes = all // hclsyntax.ScopeTraversalExpression
+	//
+	// We support the following in a static context:
+	//   ignore_changes = <static_expression>
+	//   ignore_changes = local.bar
+	// With the following forms:
+	//   <static_expression> = ["ami", "instance_type"] // hclsyntax.TulpleConsExpr *ExprList* hclsyntax.TemplateExpr
+	//   <static_expression> = "all" hclsyntax.TemplateExpr
+	// For example:
+	//   ignore_changes = var.ignore_changes // hclsyntax.ScopeTraversalExpression
+	//   ignore_changes = var.changes_overridden ? local.valid_overrides : []
+	//
+	// List comprehension is also supported (if resolvable in static context)
+	// [for x in range(10): x] hclsyntax.ForExpr
+	//
+	// We also handle two legacy forms for compatibility with earlier
+	// versions:
+	//   ignore_changes = ["ami", "instance_type"]
+	//   ignore_changes = ["*"]
+
+	// Potential inputs:
+	// * -> ERROR
+	// "*" -> ERROR
+	// all -> IgnoreAllChanges = true
+	// "all" -> IgnoreAllChanges = true
+	// [all] -> [all]
+	// ["all"] -> [all]
+	// [] -> []
+	// [foo] -> [foo]
+	// ["foo"] -> [foo]
+	// [var.bar] -> EITHER value of var.bar if it exists or var.bar traversal. This scenario is ambiguous
+	// [foo, "bar"] -> [foo, bar]
+	// [foo, var.bar] -> [foo, var.bar]
+	// ["foo", var.bar] -> [foo, ${var.bar}]
+	// concat(["foo"], ["bar"]) -> [foo, bar]
+	// concat([foo], [bar]) -> ERROR!
+
+	if hcl.ExprAsKeyword(attr.Expr) == "all" {
+		r.Managed.IgnoreAllChanges = true
+		return diags
+	}
+
+	// Try evaluating as static expression
+	ident := StaticIdentifier{Module: eval.call.addr, Subject: fmt.Sprintf("%s.%s.lifecycle.%s", r.Type, r.Name, attr.Name), DeclRange: attr.Range}
+	val, valDiags := eval.Evaluate(attr.Expr, ident)
+	if valDiags.HasErrors() {
```

Comment:
I agree with the linter here, let's try and invert this if so we're not nesting so much

Reply:
That's not easy at all, IMO this is the cleanest way to implement this.